CMakeLists.txt

```diff
@@ -57,6 +57,7 @@ if(MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3")
 else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wundef -Wshadow")
+    set(CMAKE_CXX_DEBUG_FLAGS "${CMAKE_CXX_DEBUG_FLAGS} -D_GLIBCXX_DEBUG")
```

Comment:
I don't think it should be a default setting because it slows down the test suite significantly (we discussed it a long while ago)